### ISSUE TITLE
An access_token request does not work for VCs with jwt_vc_json because https://identity.foundation/claim-format-registry/ expects jwt_vc.

### ISSUE THREAD

**Issue by rolandgroen** [nuts-foundation/nuts-node#2755]
The method RequestRFC021AccessToken filters the formatCandidates on both the metadata settings from the presentationDefinition.Format and the oauth server metadata. 

The VP metadata returns this:

```
func DefaultOpenIDSupportedFormats() map[string]map[string][]string {
	return map[string]map[string][]string{
		"jwt_vp_json": {"alg_values_supported": algValuesSupported},
		"jwt_vc_json": {"alg_values_supported": algValuesSupported},
		"ldp_vc":      {"proof_type_values_supported": proofTypeValuesSupported},
		"ldp_vp":      {"proof_type_values_supported": proofTypeValuesSupported},
	}
}
```

The presentationDefinition (note that https://identity.foundation/claim-format-registry/ does not allow `jwt_vp_json` and `jwt_vc_json`):

```   "format": {
      "ldp_vc": {
        "proof_type": [
          "JsonWebSignature2020"
        ]
      },
      "ldp_vp": {
        "proof_type": [
          "JsonWebSignature2020"
        ]
      },
      ...
    }
```

The VC that sits in the wallet is of type `jwt_vp_json`.

Problem: the VC of type `jwt_vc_json` does not get evaluated as candidate.
Expected: the VC of type `jwt_vc_json`  should be evaluated as candidate.

**Comment by rolandgroen** [comment]
Created PR: https://github.com/nuts-foundation/nuts-node/pull/2756

**Comment by gerardsn** [comment]
fixed by #2892
